flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistry.java

```diff
@@ -163,7 +170,16 @@ public void startQueryService(ActorSystem actorSystem) {
 	}
 
 	public char getDelimiter() {
-		return this.delimiter;
+		return this.globalDelimiter;
+	}
+
+	public char getDelimiter(int reporterIndex) {
+		try {
+			return delimiters.get(reporterIndex);
+		}catch (IndexOutOfBoundsException e){
+			LOG.warn("Delimiter for index {} not found return global delimiter",reporterIndex);
```

Comment:
missing space after comma. Message should be "Delimiter for index {} not found, returning global delimiter."